svd-rs/src/peripheralinfo.rs

```diff
@@ -0,0 +1,304 @@
+                .name
+                .ok_or_else(|| BuildError::Uninitialized("name".to_string()))?,
+            display_name: self.display_name.empty_to_none(),
+            version: self.version.empty_to_none(),
+            description: self.description.empty_to_none(),
+            group_name: self.group_name.empty_to_none(),
+            base_address: self
+                .base_address
+                .ok_or_else(|| BuildError::Uninitialized("base_address".to_string()))?,
+            default_register_properties: self.default_register_properties.build(lvl)?,
+            address_block: self.address_block,
+            interrupt: self.interrupt.unwrap_or_default(),
+            registers: self.registers,
+            derived_from: self.derived_from,
+        };
+        if !lvl.is_disabled() {
+            per.validate(lvl)?;
+        }
+        Ok(per)
+    }
+}
+
+impl PeripheralInfo {
+    /// Make a builder for [`Peripheral`]
+    pub fn builder() -> PeripheralInfoBuilder {
+        PeripheralInfoBuilder::default()
+    }
+    /// Modify an existing [`Peripheral`] based on a [builder](PeripheralInfoBuilder).
+    pub fn modify_from(
+        &mut self,
+        builder: PeripheralInfoBuilder,
+        lvl: ValidateLevel,
+    ) -> Result<(), SvdError> {
+        if let Some(name) = builder.name {
+            self.name = name;
+        }
+        if builder.display_name.is_some() {
+            self.display_name = builder.display_name.empty_to_none();
+        }
+        if builder.version.is_some() {
+            self.version = builder.version.empty_to_none();
+        }
+        if builder.description.is_some() {
+            self.description = builder.description.empty_to_none();
+        }
+        if builder.group_name.is_some() {
+            self.group_name = builder.group_name.empty_to_none();
+        }
+        if let Some(base_address) = builder.base_address {
+            self.base_address = base_address;
+        }
+        if let Some(interrupt) = builder.interrupt {
+            self.interrupt = interrupt;
+        }
+        if builder.derived_from.is_some() {
+            self.derived_from = builder.derived_from;
+            self.registers = None;
+            self.address_block = None;
+            self.default_register_properties = RegisterProperties::default();
+        } else {
+            if builder.address_block.is_some() {
+                self.address_block = builder.address_block;
+            }
+            self.default_register_properties
+                .modify_from(builder.default_register_properties, lvl)?;
+            if builder.registers.is_some() {
+                self.registers = builder.registers.empty_to_none();
+            }
+        }
+        if !lvl.is_disabled() {
+            self.validate(lvl)
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Validate the [`Peripheral`]
+    pub fn validate(&mut self, lvl: ValidateLevel) -> Result<(), SvdError> {
+        // TODO
```

Comment:
what is this todo? Comes from https://github.com/rust-embedded/svd/blob/81459691f379811650e7be81d81043e06b57437d/svd-rs/src/peripheral.rs#L263
But I think it was left in by mistake